R/dinucleotideFrequencyCalcandPlot.R

```diff
@@ -0,0 +1,142 @@
+## dependencies
+library("BSgenome")
+library("BSgenome.Hsapiens.UCSC.hg19")
+library("GenomicDistributions")
+library("Biostrings")
+genome <- BSgenome.Hsapiens.UCSC.hg19
+bsg = GenomicDistributions::loadBSgenome('hg19')
+queryFile = system.file("extdata", "vistaEnhancers.bed.gz", package="GenomicDistributions")
+query = rtracklayer::import(queryFile)
+## end dependencies
+
+## calc function
+
+calcDinuclFreq <- function(query, ref) {
+  GenomicDistributions:::.validateInputs(list(query=c("GRanges","GRangesList"),
+                                              ref="BSgenome"))
+   # I believe this function validates the input as GRanges
+  
+  if (is(query, "GRangesList")) {
+    # Recurse over each GRanges object
+    x = lapply(query, calcdinuclfreq, ref)
+    namelist=names(query)
+    if (is.null(namelist)){
+      newnames=seq_along(query)
+      namelist=newnames
+      
+      names(x)=namelist
+    }
+    return(x)
+  }
+  
+  
+  # This returns a list of the inputs with lapply
+  seqlevels(query, pruning.mode="coarse")=seqlevels(ref)
+  v = IRanges::Views(ref, query)
+  
+  # The Views virtual class is a general container for storing a set of views 
+  # on an arbitrary Vector object, called the "subject"
+  # The subject in this case are inputs
+  dnvec= as.data.frame(Biostrings::dinucleotideFrequency(v))
+                       
+  return(dnvec)
+ }
+
+## test function
+
+DNF<-calcDinuclFreq(query, bsg)
```

Comment:
Also `DNF<-calcDinuclFreq(query, bsg)` - line 47 should not be there. Examples are in the header above the function.

Reply:
Sorry, I noticed this problem, because I uploaded an outdated version.